core/src/autogluon/core/utils/utils.py

```diff
@@ -402,32 +402,17 @@ def compute_weighted_metric(y, y_pred, metric, weights, weight_evaluation=None,
     """ Report weighted metric if: weights is not None, weight_evaluation=True, and the given metric supports sample weights.
         If weight_evaluation=None, it will be set to False if weights=None, True otherwise.
     """
-    quantile_levels = None
-    if metric.needs_quantile:
-        quantile_levels = kwargs.pop('quantile_levels', None)
-        if quantile_levels is None:
-            raise ValueError("eval_metric='{metric.name}' requires a list of quantile levels predicting")
-
     if weight_evaluation is None:
         weight_evaluation = not (weights is None)
     if weight_evaluation and weights is None:
         raise ValueError("Sample weights cannot be None when weight_evaluation=True.")
     if not weight_evaluation:
-        if quantile_levels is not None:
-            return metric(y, y_pred, quantile_levels=quantile_levels)
-        else:
-            return metric(y, y_pred)
+        return metric(y, y_pred, **kwargs)
     try:
-        if quantile_levels is not None:
-            weighted_metric = metric(y, y_pred, quantile_levels=quantile_levels, sample_weight=weights)
-        else:
-            weighted_metric = metric(y, y_pred, sample_weight=weights)
+        weighted_metric = metric(y, y_pred, sample_weight=weights, **kwargs)
```

Comment:
I expect kwargs are going to cause crashes if we pass `quantile_levels=None` when not doing quantile regression. This will probably be evident once the CI is run. The solution is to not pass `quantile_levels=None` to `compute_weighted_metric`, but instead not pass anything at all.

Reply:
Do you think this looks good changing inside score functions?

```
if metric.needs_quantile:
    return compute_weighted_metric(y, y_pred, metric, sample_weight, quantile_levels=self.quantile_levels)
else:
    return compute_weighted_metric(y, y_pred, metric, sample_weight)
```

Reply:
I think the check whether the metric needs quantile or not should happen inside of `compute_weighted_metric`. And then you can still allow kwargs as an arg of `compute_weighted_metric`, but do not pass these kwargs into the inner `metric(...)`

Reply:
Probably this will be good? This version works properly.
```
def compute_weighted_metric(y, y_pred, metric, weights, weight_evaluation=None, **kwargs):
    """ Report weighted metric if: weights is not None, weight_evaluation=True, and the given metric supports sample weights.
        If weight_evaluation=None, it will be set to False if weights=None, True otherwise.
    """
    if not metric.needs_quantile:
        kwargs.pop('quantile_levels')
```

Reply:
That will crash if `quantile_levels` was not provided. Instead do `kwargs.pop('quantile_levels', None)`.

Reply:
got it, thanks.